web/src/beta/lib/reearth-ui/components/Modal/index.tsx

```diff
@@ -0,0 +1,65 @@
+import useTransition from "@rot1024/use-transition";
+import { FC, ReactNode, useCallback, useMemo, useRef } from "react";
+import { useClickAway, useKeyPressEvent } from "react-use";
+
+import { styled } from "@reearth/services/theme";
+
+export type ModalProps = {
+  visible: boolean;
+  children: ReactNode;
+  size?: "small" | "medium" | "large";
+  onClose?: () => void;
+};
+
+const DEFAULT_MODAL_WIDTH = 572;
+
+export const Modal: FC<ModalProps> = ({ visible, children, size, onClose }) => {
+  const ref = useRef<HTMLDivElement>(null);
+  useClickAway(ref, () => onClose?.());
+
+  const modalWidth = useMemo(
+    () => (size === "small" ? 416 : size === "large" ? 778 : DEFAULT_MODAL_WIDTH),
```

Comment:
Coule you define the value of small and large as constants as well? eg. 
MODAL_SIZE_SMALL 
MODAL_SIZE_MEDIUM 
MODAL_SIZE_LARGE